src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,3 +390,128 @@ def get_policy_assignment_completion_list(prefix, **kwargs):#pylint: disable=unu
     result = policy_client.policy_assignments.list()
     return [i.name for i in result]
 
+class _ResourceUtils(object): #pylint: disable=too-many-instance-attributes
+    def __init__(self, resource_group_name=None, resource_provider_namespace=None,
+                 parent_resource_path=None, resource_type=None, resource_name=None,
+                 resource_id=None, api_version=None, rcf=None):
+        if bool(resource_id) == bool(resource_group_name or resource_type or
+                                     parent_resource_path or resource_provider_namespace or
+                                     resource_name):
+            raise IncorrectUsageError(
+                "You must specify either 'id' or other individual pieces, but not both")
```

Comment:
You might put this in usage statement form, like the error for incorrect use of `--ids` does. So something like (--id ID | --resource-group RG --name NAME --provider ...)